tests/miral/invalid_requests.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright © 2019 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: William Wold <william.wold@canonical.com>
+ */
+
+#include "test_window_manager_tools.h"
+
+using namespace miral;
+using namespace testing;
+namespace mt = mir::test;
+using mir::operator<<;
+
+namespace
+{
+X const display_left{0};
+Y const display_top{0};
+Width const display_width{640};
+Height const display_height{480};
+
+Rectangle const display_area{{display_left,  display_top},
+                             {display_width, display_height}};
+
+struct InvalidRequests;
+
+struct InvalidParam
+{
+    std::string name;
+    std::function<Window(InvalidRequests&)> window;
+};
+
+std::ostream& operator<<(std::ostream& ostream, InvalidParam const& param)
+{
+    return ostream << param.name;
+}
+
+struct InvalidRequests : mt::TestWindowManagerTools, WithParamInterface<InvalidParam>
```

Comment:
Maybe "Invalid" is overly strong wording for requests that are valid, but will be ignored. "Obsolete"?